lib/spack/spack/test/cmd/env.py

```diff
@@ -230,8 +232,8 @@ def test_activate_adds_transitive_run_deps_to_path(
     with e:
         install('depends-on-run-env')
 
-    cmds = ev.activate(e)
-    assert 'DEPENDENCY_ENV_VAR=1' in cmds
+    env_variables = spack.environment.shell.activate(e).apply_modifications({})
+    assert env_variables['DEPENDENCY_ENV_VAR'] == '1'
```

Comment:
This is what I was referring to. Reading these two lines one could think `apply_modifications` computes a new return value, while instead it's modifying the literal `{}` passed as argument. I think we should:
```suggestion
    env_variables = {}
    spack.environment.shell.activate(e).apply_modifications(env_variables)
    assert env_variables['DEPENDENCY_ENV_VAR'] == '1'
```
to be more explicit on what is happening underneath.